api/internal/handler/plugin_config/plugin_config.go

```diff
@@ -0,0 +1,228 @@
+	r.PATCH("/apisix/admin/plugin_configs/:id", wgin.Wraps(h.Patch,
+		wrapper.InputType(reflect.TypeOf(PatchInput{}))))
+	r.PATCH("/apisix/admin/plugin_configs/:id/*path", wgin.Wraps(h.Patch,
+		wrapper.InputType(reflect.TypeOf(PatchInput{}))))
+	r.DELETE("/apisix/admin/plugin_configs/:ids", wgin.Wraps(h.BatchDelete,
+		wrapper.InputType(reflect.TypeOf(BatchDelete{}))))
+}
+
+type GetInput struct {
+	ID string `auto_read:"id,path" validate:"required"`
+}
+
+func (h *Handler) Get(c droplet.Context) (interface{}, error) {
+	input := c.Input().(*GetInput)
+
+	pluginConfig, err := h.pluginConfigStore.Get(c.Context(), input.ID)
+	if err != nil {
+		return handler.SpecCodeResponse(err), err
+	}
+
+	return pluginConfig, nil
+}
+
+type ListInput struct {
+	Search string `auto_read:"search,query"`
+	store.Pagination
+}
+
+// swagger:operation GET /apisix/admin/plugin_configs getPluginConfigList
+//
+// Return the plugin_config list according to the specified page number and page size, and support search.
+//
+// ---
+// produces:
+// - application/json
+// parameters:
+// - name: page
+//   in: query
+//   description: page number
+//   required: false
+//   type: integer
+// - name: page_size
+//   in: query
+//   description: page size
+//   required: false
+//   type: integer
+// - name: search
+//   in: query
+//   description: search keyword
+//   required: false
+//   type: string
+// responses:
+//   '0':
+//     description: list response
+//     schema:
+//       type: array
+//       items:
+//         "$ref": "#/definitions/pluginConfig"
+//   default:
+//     description: unexpected error
+//     schema:
+//       "$ref": "#/definitions/ApiError"
+func (h *Handler) List(c droplet.Context) (interface{}, error) {
+	input := c.Input().(*ListInput)
+
+	ret, err := h.pluginConfigStore.List(c.Context(), store.ListInput{
+		Predicate: func(obj interface{}) bool {
+			if input.Search != "" {
+				return strings.Contains(obj.(*entity.PluginConfig).Desc, input.Search)
+			}
+			return true
+		},
+		PageSize:   input.PageSize,
+		PageNumber: input.PageNumber,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return ret, nil
+}
+
+func (h *Handler) Create(c droplet.Context) (interface{}, error) {
+	input := c.Input().(*entity.PluginConfig)
+
+	ret, err := h.pluginConfigStore.Create(c.Context(), input)
+	if err != nil {
+		return handler.SpecCodeResponse(err), err
+	}
+
+	return ret, nil
+}
+
+type UpdateInput struct {
+	ID string `auto_read:"id,path"`
+	entity.PluginConfig
+}
+
+func (h *Handler) Update(c droplet.Context) (interface{}, error) {
+	input := c.Input().(*UpdateInput)
+
+	// check if ID in body is equal ID in path
+	if err := handler.IDCompare(input.ID, input.PluginConfig.ID); err != nil {
+		return &data.SpecCodeResponse{StatusCode: http.StatusBadRequest}, err
+	}
+
+	if input.ID != "" {
+		input.PluginConfig.ID = input.ID
+	}
+
+	ret, err := h.pluginConfigStore.Update(c.Context(), &input.PluginConfig, true)
```

Comment:
Please make sure about the third args in `h.pluginConfigStore.Update`.
We should create it if not exist?

Reply:
> Please make sure about the third args in `h.pluginConfigStore.Update`.
> We should create it if not exist?

sure, we support create by `PUT`

Reply:
OK.Got it.